leelastar/network/layers.py

```diff
@@ -0,0 +1,76 @@
+"""
+@file: layers.py
+Created on 02.02.19
+@project: LeelaStar
+@author: kiudee
+
+Groups the defined basic layer types used for the project
+"""
+from collections import OrderedDict
+import torch.nn as nn
+
+
+class SqueezeExcitation(nn.Module):
+    """Construct a squeeze excitation block for use in a residual block.
+
+    This uses the squeeze excitation block as proposed by `Hu et al. 2017`_.
+
+    .. _Hu et al. 2017:
+        https://arxiv.org/abs/1709.01507
+
+    """
+
+    def __init__(self, channels, ratio):
+        super().__init__()
+
+        self.global_pooling = nn.AdaptiveAvgPool2d(1)
+        self.dense_linear_1 = nn.Linear(channels, channels // ratio)
+        self.relu = nn.ReLU(inplace=True)
```

Comment:
typically you'd only pre-define layers with trained parameters (so no need for `global_pooling`) either

Reply:
The pooling is part of the squeeze excitation mechanism.

Reply:
Sure, it still doesn't have any trainable parameters